core/src/main/java/hudson/model/JDK.java

```diff
@@ -183,8 +185,15 @@ public String getDisplayName() {
         }
 
         @Override
-        public List<JDKInstaller> getDefaultInstallers() {
-            return Collections.singletonList(new JDKInstaller(null,false));
+        public List<? extends ToolInstaller> getDefaultInstallers() {
```

Comment:
I think this is the only change that will create a binary incompatibility.

Reply:
Is it really? Won't erasure take care of this?

Reply:
Yes you're right, I think the relevant section of the spec is [jls-13.4.15](https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.15). It will create a _source_ incompatibility though, right?

Reply:
Yes, but it is binary compatible. I think this is fine. Anyway it seems unlikely plugins are referring to the method definition in the supertype, much less this override.